docs/source/guides/data.rst

```diff
@@ -0,0 +1,562 @@
+defined within your :class:`~pytorch_lightning.core.Lightning.LightningModule` instance. For example, rhis would be the case if your test data
+set is not available at the time your model was declared. Simply pass the test set to the :meth:`~pytorch_lightning.trainer.trainer.Trainer.Trainer.test` method:
+
+.. code-block:: python
+
+    # setup your data loader
+    test = DataLoader(...)
+
+    # test (pass in the loader)
+    trainer.test(test_dataloaders=test)
+
+--------------
+
+
+.. _sequences:
+
+
+***************
+Sequential Data
+***************
+
+Lightning has built in support for dealing with sequential data.
+
+
+Packed sequences as inputs
+==========================
+When using PackedSequence, do 2 things:
+
+1. Return either a padded tensor in :class:`~torch.utils.data.Dataset` or a list of variable length tensors in the :class:`~torch.utils.data.DataLoader` collate_fn (example shows the list implementation).
+2. Pack the sequence in forward or training and validation steps depending on use case.
+
+.. testcode::
+
+    # For use in DataLoader
+    def collate_fn(batch):
+        x = [item[0] for item in batch]
+        y = [item[1] for item in batch]
+        return x, y
+
+    # In module
+    def training_step(self, batch, batch_nb):
+        x = rnn.pack_sequence(batch[0], enforce_sorted=False)
+        y = rnn.pack_sequence(batch[1], enforce_sorted=False)
+
+----------
+
+Truncated Backpropagation Through Time
+======================================
+There are times when multiple backwards passes are needed for each batch.
+For example, it may save memory to use Truncated Backpropagation Through Time when training RNNs.
+
+Lightning can handle TBTT automatically via this flag.
+
+.. testcode::
+
+    from pytorch_lightning import LightningModule
+
+    class MyModel(LightningModule):
+
+        def __init__(self):
+            super().__init__()
+            # Important: This property activates truncated backpropagation through time
+            # Setting this value to 2 splits the batch into sequences of size 2
+            self.truncated_bptt_steps = 2
+
+        # Truncated back-propagation through time
+        def training_step(self, batch, batch_idx, hiddens):
+            # the training step must be updated to accept a ``hiddens`` argument
+            # hiddens are the hiddens from the previous truncated backprop step
+            out, hiddens = self.lstm(data, hiddens)
+            return {
+                "loss": ...,
+                "hiddens": hiddens
+            }
+
+.. note:: If you need to modify how the batch is split,
+    override :meth:`~pytorch_lightning.core.LightningModule.tbptt_split_batch`.
+
+----------
+
+Iterable Datasets
+=================
+Lightning supports using IterableDatasets as well as map-style Datasets. IterableDatasets provide a more natural
+option when using sequential data.
```

Comment:
this feels odd at the very end of the guide. iterable datasets should be covered as part of the pytorch data reading. does it need to be repeated again here?